web_poet/exceptions/http.py

```diff
@@ -73,3 +73,20 @@ def __init__(
         if msg is None:
             msg = f"Unexpected HTTP Response received: {self.response}"
         super().__init__(msg, request=request)
+
+
+class NoSavedHttpResponse(HttpError):
```

Comment:
Hm, I'm not sure it's good to have it as a subclass of HttpError. For example, if a page object handles HttpErrors (which is perfectly valid), then we might not get a test failure, as the page object might "handle" this NoSavedHttpResponse.

Should it be a subclass of AssertionError?

Reply:
Yeah, I wasn't sure if this counts as HttpError or not. 

I don't know about AssertionError, will it have some consequences, e.g. special handling by pytest?

Reply:
Should it also be moved to exceptions/core.py?

Reply:
We also have testing/exceptions.py, though I'm not sure it's good that they're not in exceptions/testing.py.

Reply:
What do you think about moving testing/exceptions.py to exceptions/testing.py, and then moving NoSavedHttpResponse there?

Reply:
These exceptions are not the ones which users should be catching though; that's why I've added them to testing module, not to top-level web_poet.exceptions. `exceptions/_testing.py` is also an option.

Reply:
Yeah, so it makes sense to leave `testing/exceptions.py` as is, then what should we do with `NoSavedHttpResponse`?

Reply:
Options:

1. Move it to `testing/exceptions.py`.
2. Keep it as-is.
3. Move to core.py.

Either way, I think it shouldn't inherit from HttpError, as it'd lead to issues. I'm fine with any of these locations, as soon as it's not a subclass of HttpError.

We shouldn't expose this exception in `__init__.py`, because this exception is not supposed to be caught by web-poet users' code.

Reply:
Moving it to `testing/exceptions.py` causes some circular import between `web_poet.testing.fixture` and `web_poet`.

On the other hand, keeping it in `http` or moving to `core` means we need to change the current `from .core import *; from .http import *` code if we don't want to import this one there :)

Reply:
> On the other hand, keeping it in http or moving to core means we need to change the current from .core import *; from .http import * code if we don't want to import this one there :)

Right; using `__all__` is also an option.